### ISSUE TITLE
Additional tarball for missing content needed by distributions

### ISSUE THREAD

**Issue by iskunk** [chromium-linux-tarballs/chromium-tarballs#18]
We would like our primary Chromium tarball to match the upstream `-lite` tarball, ideally being bit-for-bit identical. This should now be feasible given the [recent update](https://chromium-review.googlesource.com/c/chromium/tools/build/+/6083831) to the `export_tarball.py` script.

However, Debian cannot make use of the `-lite` or equivalent tarball, because their build needs the bundled version of ICU (under `third_party/icu/`) which is excluded from `-lite`. The latest version of ICU available in Debian is 72, and the bundled version is 74. (The current version is 76 as of this writing.)

What I'd like to propose is a new tarball, most apropos suffix I can think of is `-linux-extra`, that would include `third_party/icu/` and in general anything that some Linux distribution needs for their build that is absent from our tarball (yet present in Google's full-fat one). This should allow us to generate a primary tarball that exactly matches upstream's `-lite`, while still being able to cater to the additional requirements of individual distributions, saving them the trouble of needing to download a much larger tarball.

Of course, anything missing from `-lite` that is needed by a broad array of distributions would motivate a change to the upstream `publish_tarball.py` script that defines the content. This tarball is only intended to address issues idiosyncratic to the organizations consuming it, with no expectations for Chromium/Google proper.

If and when Debian updates their version of ICU such that they can build Chromium with the system library, and thus no longer needs this tarball---and assuming no other distribution does either---we could disable generating it. Consumers would then need to be aware that the tarball may not always be present for a release, of course.

**Comment by iskunk** [comment]
An update:

I've been trying out upstream's lite tarball in the context of the ungoogled-chromium Windows build. I've found that there are a few, relatively small-ish things that are excluded from the lite tarball (mostly third-party library source) that are needed, and a couple large things currently present (`rust-toolchain` and `VK-GL-CTS`) that can be removed. Notably, `third_party/icu` is needed, not because of a version issue, but because no other build of the library is immediately available.

So I'm going to prepare a new upstream change that further reduces the size of the tarball while adding these library dependencies back in. That should render the above proposal unnecessary.

**Comment by Kangie** [comment]
I like your plans.

If the rust removal is big enough we should look at also publishing our tarballs as GH artefacts, for stable (131) we're only 88M above the threshold...

**Comment by iskunk** [comment]
I believe that will be possible! I just recently landed two relevant upstream changes: [6154800](https://chromium-review.googlesource.com/c/chromium/tools/build/+/6154800) and [6162014](https://chromium-review.googlesource.com/c/chromium/tools/build/+/6162014).

The lite tarball should now be in the vicinity of 1.3 GB (last I checked), and ICU is now in there. I have one more upstream change to push, mainly dealing with how files are pruned from the tree, and then will circle back here to update our scripts.

**Comment by Kangie** [comment]
I think we can close this one off now, mission accomplished?

**Comment by iskunk** [comment]
Yep, indeed this issue has been licked entirely. We don't need an additional tarball, we can just push an upstream fix to the lite tarball build. Having the lite tarball contain all source code needed in a typical Chromium build, but none of the third-party tooling, is a goal that both sides can agree upon. It works for us in Linux-land, and it works for ungoogled-chromium, too.

Also, the `.hashes` filename issue is now [fixed](https://chromium-review.googlesource.com/c/chromium/tools/build/+/6249430) ;-)
